### DISCUSSION TITLE
Need help adding custom attribute in the tags

### DISCUSSION THREAD

**Discussion by Gurushesh-Metapercept** [dita-ot/dita-ot#4505] [Q&A]
I have a DITA source where some tags have the attribute demo-data="1" (see the example below). When I try to transform the DITA to HTML5 using the default HTML5 transform type, I encounter an error like `Attribute "demo-data" must be declared for element type "p"`. My question is, how can I ensure that any tag with the attribute demo-data="1" is copied to the corresponding HTML output tag in the HTML5 plugin?

Example Input:- 
`<p demo-data="1" class="- topic/p ">Engage Home</p>`

Expected output:-
`<p demo-data="1" class="p">Engage Home</p>`

_also value of demo-data attribute can be anything_

**Comment by jelovirt** [comment]
You have two issues here:

1. You have to modify the DTD/XSD/RNG schema that you use in the topics to include the new attribute. DITA schemas don't allow for attributes that haven't been declared explicitly in the schema.
2. By default, you can use a DITAVAL filter that uses `passthrough` to copy the attribute values into HTML but those attributes will get `data-` prefix. You'd need to write a plugin to DITA-OT that allow passing through attributes as is.
